state/protocol/badger/mutator_test.go

```diff
@@ -556,8 +559,13 @@ func TestExtendEpochTransitionValid(t *testing.T) {
 		require.Error(t, err)
 
 		// finalize block 3 so we can finalize subsequent blocks
+		// ensure an epoch phase transition when we finalize block 3
+		consumer.On("EpochSetupPhaseStarted", epoch2Setup.Counter-1, block3.Header).Once()
+		metrics.On("CurrentEpochPhase", flow.EpochPhaseSetup)
 		err = state.Finalize(block3.ID())
 		require.NoError(t, err)
+		consumer.AssertCalled(t, "EpochSetupPhaseStarted", epoch2Setup.Counter-1, block3.Header)
+		metrics.AssertCalled(t, "CurrentEpochPhase", flow.EpochPhaseSetup)
```

Comment:
I think this timing is off. See the comments on line 570/571: the phase transition occurs when we finalize block 4. There is already an assertion about the phase transition event on line 572, we should put the assertion about the metric call at the same time (since they happen at the same time in the implementation).